### PR TITLE
More logging for the missing audit entries investigation.

### PR DESCRIPTION
**Pull request by agr**
Should help with the investigation for the [eng-#492](https://github.com/NuGet/Engineering/issues/492).

### CODE REVIEW

File: src/Validation.Common/PackageValidationAuditor.cs [hunk 1]
```diff
@@ -131,31 +142,45 @@ public PackageValidationAuditor(CloudStorageAccount cloudStorageAccount, string
...
                 packageValidationAudit = updateAudit(packageValidationAudit);
             }
             else
             {
+                _logger.LogInformation($"Updating new blob for validation {{{TraceConstant.ValidationId}}}",
+                    validationId);
                 packageValidationAudit = updateAudit(null);
             }
 
+            _logger.LogInformation($"Uploading blob for validation {{{TraceConstant.ValidationId}}}",
```
Comment (dtivel) [inline src/Validation.Common/PackageValidationAuditor.cs]: Can you further distinguish this log event from the previous two?  Perhaps:  "Uploading text for validation ..."

File: src/Validation.Common/PackageValidationAuditor.cs [hunk 2]
```diff
@@ -131,31 +142,45 @@ public PackageValidationAuditor(CloudStorageAccount cloudStorageAccount, string
...
                 try
                 {
+                    _logger.LogInformation($"Releasing lease for validation {{{TraceConstant.ValidationId}}}",
+                        validationId);
                     await blob.ReleaseLeaseAsync(AccessCondition.GenerateLeaseCondition(leaseId));
                 }
                 catch
                 {
                     // intentional, lease may already have been expired
+                    _logger.LogInformation($"Exception occurred while releasing the lease for validation {{{TraceConstant.ValidationId}}}",
```
Comment (dtivel) [inline src/Validation.Common/PackageValidationAuditor.cs]: Should we log the exception too?

File: src/Validation.Common/PackageValidationAuditor.cs [hunk 3]
```diff
@@ -131,31 +142,45 @@ public PackageValidationAuditor(CloudStorageAccount cloudStorageAccount, string
...
+                        validationId);
                 }
             }
+            _logger.LogInformation($"Done blob operations for validation {{{TraceConstant.ValidationId}}}",
```
Comment (dtivel) [inline src/Validation.Common/PackageValidationAuditor.cs]: Done => Completed

File: src/Validation.Common/PackageValidationAuditor.cs [hunk 4]
```diff
@@ -116,13 +116,24 @@ public PackageValidationAuditor(CloudStorageAccount cloudStorageAccount, string
 
         public async Task StoreAuditAsync(Guid validationId, string packageId, string packageVersion, Func<PackageValidationAudit, PackageValidationAudit> updateAudit)
         {
+            _logger.LogInformation($"Updating audit blob for validation {{{TraceConstant.ValidationId}}} " +
```
Comment (scottbommarito) [inline src/Validation.Common/PackageValidationAuditor.cs]: Maybe `Started storing audit blob for validation` instead?

File: src/Validation.Common/PackageValidationAuditor.cs [hunk 5]
```diff
@@ -131,31 +142,48 @@ public PackageValidationAuditor(CloudStorageAccount cloudStorageAccount, string
...
             {
+                _logger.LogInformation($"Updating existing auditing blob for validation {{{TraceConstant.ValidationId}}}",
+                    validationId);
                 var json = await blob.DownloadTextAsync(Encoding.UTF8, accessCondition, null, null);
                 packageValidationAudit = JsonConvert.DeserializeObject<PackageValidationAudit>(json);
                 packageValidationAudit = updateAudit(packageValidationAudit);
             }
             else
             {
+                _logger.LogInformation($"Updating new auditing blob for validation {{{TraceConstant.ValidationId}}}",
```
Comment (scottbommarito) [inline src/Validation.Common/PackageValidationAuditor.cs]: `Creating new auditing blob`?

The blob doesn't exist yet, so you're not updating it.

File: src/Validation.Common/PackageValidationAuditor.cs [hunk 6]
```diff
@@ -131,31 +142,48 @@ public PackageValidationAuditor(CloudStorageAccount cloudStorageAccount, string
...
                 {
                     // intentional, lease may already have been expired
+                    _logger.LogInformation(
+                        TraceEvent.AuditBlobLeaseReleaseFailed,
+                        e,
+                        $"Exception occurred while releasing the lease for validation {{{TraceConstant.ValidationId}}}",
+                        validationId);
                 }
             }
+            _logger.LogInformation($"Completed blob operations for validation {{{TraceConstant.ValidationId}}}",
```
Comment (scottbommarito) [inline src/Validation.Common/PackageValidationAuditor.cs]: `Completed storing audit blob for validation`?

Basically, the first message and this message need to be very similar 
(`Started X` and `Completed X`).

File: src/Validation.Common/PackageValidationAuditor.cs (hunk 5 above)
Comment (agr) [inline src/Validation.Common/PackageValidationAuditor.cs]: Not really :) There is a `updateAudit` function passed that is supposed to figure out what to do in this case (when `null` is passed to is as an argument). So `StoreAuditAsync` method does not really care what happens inside of it.

**Pull request merged**